crates/subspace-networking/src/node_runner.rs

```diff
@@ -397,10 +398,16 @@ where
                 .kademlia
                 .remove_peer(&event.peer_id);
         } else {
+            // Remove both versions of the address
```

Comment:
Is it really possible to have both? It seems odd to not have identity in the address, how would you establish secure connection to someone you don't know identity of?

Reply:
I wouldn't handle it then if it is not used

Reply:
Well, I'm not sure when we use address without Peer ID, this is why I'm confused.

Reply:
We use addresses without Peer ID when we add them as self-reported addresses to Kademlia. They become self-reported when they are taken from listening and external addresses lists on the other peer.

Reply:
But we do not add our own address to our Kademlia, do we? I don't think I follow you here.

Reply:
Actually we do (implicitly) - when we add external addresses and specify listening addresses. When we request providers - Kademlia fills providers with those addresses (both lists chained).